roles/openshift_health_checker/openshift_checks/diskspace_availability.py

```diff
@@ -0,0 +1,115 @@
+        self.minimum_docker_space_percent = get_var(task_vars, "healthcheck_minimum_docker_space_percent", default=self.minimum_docker_space_percent)
+
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+
+        # if running containerized installation check that available
+        # space is not below defined minimum threshold
+        if is_containerized:
+            failed, msg = self.containerized_volume_check(task_vars)
+            if failed:
+                return {"failed": True, "msg": msg}
+
+        failed, msg = self.noncontainerized_volume_check(ansible_mounts, task_vars)
+        return {"failed": failed, "msg": msg}
+
+    def containerized_volume_check(self, task_vars):
+        docker_facts = self.module_executor("docker_facts", {}, task_vars)
+        info = docker_facts.get("docker_facts")
+        status = dict(info.get("DriverStatus"))
+
+        # ensure that the docker host is using a storage driver that our installation
+        # is known to support. The only supported driver is currently "devicemapper"
+        if info.get("Driver", "") == self.supported_docker_dev_driver:
+            disktotal = self.to_bytes(status.get("Data Space Total", ""))
+            diskavail = self.to_bytes(status.get("Data Space Available", ""))
+
+            if not disktotal or not diskavail:
+                msg = "Unable to retrieve storage driver information for {driver}.".format(driver=self.supported_docker_dev_driver)
+                return True, msg
+
+            diskleft_percent = diskavail / disktotal * 100
+
+            # fail if available diskspace is less than minimum docker threshold
+            if diskleft_percent < self.minimum_docker_space_percent:
+                msg = "The amount of data space remaining ({remaining}%%) for the docker storage driver is below minimum%%".format(remaining=diskleft_percent, minimum=self.minimum_docker_space_percent)
+                return True, msg
+
+    def noncontainerized_volume_check(self, ansible_mounts, task_vars):
+        recommended_diskspace_gb = self.get_recommended_diskspace_gb(task_vars)
+        diskfree_openshift = self.get_openshift_disk_availability(ansible_mounts)
+
+        if self.to_gigabytes(diskfree_openshift) < recommended_diskspace_gb:
+            msg = "Available disk space ({diskfree} GB) for OpenShift volume below recommended storage. Minimum required disk space: {recommended} GB".format(diskfree=self.to_gigabytes(diskfree_openshift), recommended=recommended_diskspace_gb)
+            return True, msg
+
+        return False, ""
+
+    def get_recommended_diskspace_gb(self, task_vars):
+        group_names = get_var(task_vars, "group_names", default=[])
+        if "masters" in group_names:
+            return self.recommended_master_space_gb
+
+        return self.recommended_node_space_gb
```

Comment:
This is also assuming that everything that is not a master is a node. If we expect to run this check on non-master-non-node hosts, we should follow the docs and use the proper storage required for each type of host.